projects/packages/post-list/src/class-post-list.php

```diff
@@ -153,6 +153,53 @@ public function add_filters_and_actions_for_screen( $current_screen ) {
 
 		$this->maybe_customize_columns( $current_screen->post_type );
 		$this->maybe_add_share_action( $current_screen->post_type );
+		$this->maybe_add_copy_link_action( $current_screen->post_type );
+	}
+
+	/**
+	 * Checks the current post type and adds the Copy link post
+	 * action if it is appropriate to do so.
+	 *
+	 * @param string $post_type The post type associated with the current request.
+	 */
+	public function maybe_add_copy_link_action( $post_type ) {
+		if ( is_post_type_viewable( $post_type ) ) {
+			wp_add_inline_script(
+				'common',
+				'function copyLinkQuickAction( event ) {
+					event.preventDefault();
+					window.navigator.clipboard.writeText( event.target.getAttribute("href") ).then(() => {
+						event.target.textContent = "' . esc_js( __( 'Copied!', 'jetpack-post-list' ) ) . '";
+						setTimeout(() => {
+							event.target.textContent = "' . esc_js( __( 'Copy link', 'jetpack-post-list' ) ) . '";
+						}, 2000);
+					});
+				}'
+			);
```

Comment:
Instead of creating an inline script, does it make sense to move this to a separate JS file and enqueue it together with the CSS here:

https://github.com/Automattic/jetpack/blob/6098b295664077cdde7d9acc16b2ad3fc9168836/projects/packages/post-list/src/class-post-list.php#L83C4-L87

Reply:
Probably not as is right now, because `enqueue_scripts` is called even if we don't add any quick action. So it's not needed.

Reply:
If we followed that logic, we'd inline a major percentage of our JS, but we don't do that. Not to mention the downsides and lack of readability with inline JS. Is there a really good reason to prefer inline JS really?